web/modules/custom/portland/modules/portland_location_picker/js/portland_location_picker.js

```diff
@@ -810,34 +810,65 @@
         // HELPER FUNCTIONS ///////////////////////////////
 
         function handleCityLimits(latlng, muniLayer = municipalitiesLayer) {
-          // we want to call this late in the event handling process, so that previously collected coordinates
-          // or address values are cleared first.
-
-          if (requireCityLimits && displayCityLimits) {
-            // check if click is within Portland city limits. if not, use js alert to show error message and return null.
-            var inLayer = leafletPip.pointInLayer(latlng, muniLayer, false);
-            if (inLayer.length > 0) {
-              var municipalityName = inLayer[0].feature.properties.CITYNAME;
-              if (municipalityName != "Portland") {
-                console.log("Clicked in " + municipalityName + ".");
-                showStatusModal(CITY_LIMITS_MESSAGE);
-                return false;
+          if (requireCityLimits || requireCityLimitsPlusParks) {
+            var success = false;
+            $.ajax({
+              async: false,
+              url: buildCheckCityLimitsFenceUrl(latlng),
+              success: function (cityBoundaryResponse) {
+                cityBoundaryResponse = JSON.parse(cityBoundaryResponse);
+                if (!cityBoundaryResponse.features || cityBoundaryResponse.features.length < 1 || cityBoundaryResponse.features[0].attributes.CITYNAME.toLowerCase() != "portland") {
+                  // not in city boundary. if configured, try parks boundaries.
+                  if (requireCityLimitsPlusParks) {
```

Comment:
It looks like there was a small bug introduced here: if `require_city_limits` is true but `require_city_limits_plus_parks` is false, and the user clicks outside of the boundary, the "outside of city limits" modal is not displayed.

We should probably add an else {} here to `showStatusModal(CITY_LIMITS_MESSAGE)` if the first city limit check failed and plus_parks is false.

Reply:
Fixed.